packages/core/src/lib/connection_manager.ts

```diff
@@ -89,6 +89,34 @@ export class ConnectionManager
     return instance;
   }
 
+  stop(): void {
+    this.keepAliveManager.stopAll();
+    this.libp2p.removeEventListener(
+      "peer:connect",
+      this.onEventHandlers["peer:connect"]
+    );
+    this.libp2p.removeEventListener(
+      "peer:disconnect",
+      this.onEventHandlers["peer:disconnect"]
+    );
+    this.libp2p.removeEventListener(
+      "peer:discovery",
+      this.onEventHandlers["peer:discovery"]
+    );
+  }
+
+  async dropConnection(peerId: PeerId): Promise<void> {
+    try {
+      this.keepAliveManager.stop(peerId);
```

Comment:
do we need to check for case when it is last peer manager was listening to? seems no but want to clarify

Reply:
so given that in the pool there was one peer and `dropConnection` is called - do we need to do additional clean ups?